grunt prepare: fix error output?

Currently, we have the below when something bad happens. Can we see better output?

    Installing jquery-ui npm modules
    >> Error installing npm modules: undefined

I figured out that the `log` function needs to use the entire error argument:

```diff
Gruntfile.js
diff --git a/Gruntfile.js b/Gruntfile.js
index a4e72d9..7ba515c 100644
--- a/Gruntfile.js
+++ b/Gruntfile.js
@@ -102,7 +102,7 @@ grunt.initConfig({
 function log( callback, successMsg, errorMsg ) {
     return function( error, result, code ) {
         if ( error && errorMsg ) {
-            grunt.log.error( errorMsg + ": " + error.stderr );
+            grunt.log.error( errorMsg + ": " + error );
         } else if ( ! error && successMsg ) {
             grunt.log.ok( successMsg );
         }
```

With that change in place, I get to see an actual error:


```sh
Checking out jquery-ui branch/tag: 1.10.4
>> Done with checkout
Checking out api.jqueryui.com branch/tag: 1-10
>> Done with checkout
Installing jquery-ui npm modules
>> Error installing npm modules: Error: npm ERR! Darwin 13.2.0
>> npm ERR! argv "node" "/usr/local/bin/npm" "install"
>> npm ERR! node v0.10.28
>> npm ERR! npm  v2.1.4
>> npm ERR! code ETARGET
>>
>> npm ERR! notarget No compatible version found: phantomjs@'>=1.8.1 <1.9.0'
>> npm ERR! notarget Valid install targets:
>> npm ERR! notarget ["0.0.1","0.0.2","0.0.3","0.0.4","0.0.5","0.0.6","0.0.7","0.0.8","0.0.9","0.1.0","0.1.1","0.2.0","0.2.1","0.2.2","0.2.3","0.2.4","0.2.5","0.2.6","1.8.0-1","1.8.1-1","1.8.1-2","1.8.1-3","1.8.2-0","1.8.2-1","1.8.2-2","1.9.0-0","1.9.0-1","1.9.0-2","1.9.0-3","1.9.0-4","1.9.0-5","1.9.0-6","1.9.1-0","1.9.1-2","1.9.1-3","1.9.1-4","1.9.1-5","1.9.1-6","1.9.1-7","1.9.1-8","1.9.1-9","1.9.2-0","1.9.2-1","1.9.2-2","1.9.2-3","1.9.2-4","1.9.2-5","1.9.2-6","1.9.6-0","1.9.7-1","1.9.7-3","1.9.7-4","1.9.7-5","1.9.7-6","1.9.7-7","1.9.7-8","1.9.7-9","1.9.7-10","1.9.7-11","1.9.7-12","1.9.7-13","1.9.7-14","1.9.7-15","1.8.2-3","1.9.8","1.9.9","1.9.10","1.9.11","1.9.12","1.9.13","1.9.15"]
>> npm ERR! notarget
>> npm ERR! notarget This is most likely not a problem with npm itself.
>> npm ERR! notarget In most cases you or one of your dependencies are requesting
>> npm ERR! notarget a package version that doesn't exist.
>>
>> npm ERR! Please include the following file with any support request:
>> npm ERR!     /Users/jza/dev/download.jqueryui.com/tmp/jquery-ui/npm-debug.log
```

I haven't yet figured out what the issue is here. Was a published phantomjs version removed from the registry? Does newer npm use newer semver that deals with these version ranges differently? As far as I can tell, any of the `1.8.x` version except for `1.8.0` should match `phantomjs@'>=1.8.1 <1.9.0'`, but it doesn't. Can be tested separately as well:

```sh
npm install phantomjs@'>=1.8.0 <1.9.0'
```

Produces the same error. I'm started with npm `2.1.4`, then installed the latest, `2.4.1`, same issue.

--- a/Gruntfile.js
+++ b/Gruntfile.js
@@ -102,7 +102,7 @@ grunt.initConfig({
 function log( callback, successMsg, errorMsg ) {
 	return function( error, result, code ) {
 		if ( error && errorMsg ) {
-			grunt.log.error( errorMsg + ": " + error.stderr );
+			grunt.log.error( errorMsg + ": " + error );
 		} else if ( ! error && successMsg ) {
 			grunt.log.ok( successMsg );
 		}

--- a/package.json
+++ b/package.json
@@ -1,6 +1,10 @@
 {
   "name": "download.jqueryui.com",
   "version": "2.0.21",
+  "repository": {
+    "type" : "git",
+    "url" : "https://github.com/jquery/download.jqueryui.com.git"
+  },
   "dependencies": {
     "archiver": "0.4.1",
     "async": "0.1.22",
@@ -33,9 +37,11 @@
     "nodeunit": "0.7.4"
   },
   "main": "main.js",
-  "engine": {
-    "node": ">=0.8.x"
+  "engines": {
+    "node": ">=0.8.x",
+    "npm": "<2.0.0"
   },
+  "engineStrict": true,
   "licenses": [
     {
       "type": "MIT",

